ci/ci/templates/index.html

```diff
@@ -118,6 +119,15 @@ <h3>PRs</h3>
           <td align="left">
             {{ pr.author }}
           </td>
+          <td align="left">
+            {% if 'failure' == pr.build_state %}
+            <form action="/invalidate_and_retest" method="post">
```

Comment:
This needs csrf authorization. See the cancel button in batch for an example. You need the `check_csrf_token` decorator.

Reply:
done